code/modules/ruins/lavalandruin_code/sin_ruins.dm

```diff
@@ -147,6 +147,8 @@
 		return
 	if(ishuman(AM))
 		var/mob/living/carbon/human/H = AM
+		if(HAS_TRAIT(H, TRAIT_GENELESS))
+			return
```

Comment:
I'd consider adding feedback message for the user here on why the knife is not working (that their target lacks DNA). This is an active action (stabbing a mob with a knife), it might feel weird for the user if nothing happened.